### PR TITLE
Add mesh names to slicing message

### PR DESCRIPTION
**Pull request by paukstelis**
This pull requests includes the name of each mesh in the message passed to CuraEngine. This will allow including comments in slicing to delineate which blocks of gcode are associated with specific meshes.

### CODE REVIEW

**Comment by fieldOfView** [comment]
> This will allow including comments in slicing to delineate which blocks of gcode are associated with specific meshes.

Which, in turn, would allow a smart print host or firmware to skip printing that model while continuing with the rest. See eg https://plugins.octoprint.org/plugins/cancelobject/

File: plugins/CuraEngineBackend/StartSliceJob.py [hunk 1]
```diff
@@ -256,7 +256,7 @@ def run(self) -> None:
                     mesh_data = object.getMeshData()
                     rot_scale = object.getWorldTransformation().getTransposed().getData()[0:3, 0:3]
                     translate = object.getWorldTransformation().getData()[:3, 3]
-
+
```
Comment (fieldOfView) [inline plugins/CuraEngineBackend/StartSliceJob.py]: Code style: remove trailing spaces

**Comment by nallath** [comment]
For our reference; CURA-5664

File: cura/ObjectsModel.py [hunk 2]
```diff
@@ -63,6 +67,18 @@ def _update(self, *args):
...
                 is_outside_build_area = False
+                
+            #check if we already have an instance of the object based on name
+            duplicate = False
+            for n in namecount:
+            	if name == n["name"]:
+            		name = "{0}({1})".format(name, n["count"])
+            		node.setName(name)
+            		n["count"] = n["count"]+1
+            		duplicate = True
```
Comment (Ghostkeeper) [inline cura/ObjectsModel.py]: Some mixed tabs and spaces here. Could you turn that into spaces?

File: plugins/CuraEngineBackend/Cura.proto [hunk 3]
```diff
@@ -29,6 +29,7 @@ message Object
     bytes normals = 3; //An array of 3 floats.
     bytes indices = 4; //An array of ints.
     repeated Setting settings = 5; // Setting override per object, overruling the global settings.
+    string name = 6;
```
[the remark this answers is not in the document]
Comment (paukstelis) [reply plugins/CuraEngineBackend/Cura.proto]: This has been done: https://github.com/Ultimaker/CuraEngine/pull/828

File: cura/ObjectsModel.py [hunk 4]
```diff
@@ -40,6 +40,9 @@ def _update(self, *args):
         filter_current_build_plate = Application.getInstance().getPreferences().getValue("view/filter_current_build_plate")
         active_build_plate_number = self._build_plate_number
         group_nr = 1
+        instance = 1
+        namecount = []
```
[the remark this answers is not in the document]
Comment (paukstelis) [reply cura/ObjectsModel.py]: I've done this as suggested, though I'm not sure I like how either method does the numbering extension. One thing that I did notice in master branch was that after an object has been multiplied and the mesh is reloaded through the automatic reload setting, only the original object is updated. Might need to add filename/path to the node for reloading.

**Pull request merged**